addons/point_of_sale/static/src/utils.js

```diff
@@ -66,3 +66,25 @@ export function random5Chars() {
 export function qrCodeSrc(url, { size = 200 } = {}) {
     return `/report/barcode/QR/${encodeURIComponent(url)}?width=${size}&height=${size}`;
 }
+
+/**
+     * Loading image is converted to a Promise to allow await when
+     * loading an image. It resolves to the loaded image if successful,
+     * else, resolves to false.
+     *
+     * [Source](https://stackoverflow.com/questions/45788934/how-to-turn-this-callback-into-a-promise-using-async-await)
+     */
+export function loadImage(url) {
+    return new Promise((resolve) => {
+        const img = new Image();
+        img.addEventListener("load", () => resolve(img));
+        img.addEventListener("error", () => {
+            this.popup.add(ErrorPopup, {
+                title: _t("Loading Image Error"),
+                body: _t("Encountered error when loading image. Please try again."),
+            });
```

Comment:
This will crash. maybe provide add one argument `options?: { onError?: () => void }`. And if `onError` is provided, you call it in the "error" handler.